compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/IncrementalJvmCompilerRunner.kt

```diff
@@ -478,12 +478,12 @@ open class IncrementalJvmCompilerRunner(
         return exitCode to sourcesToCompile
     }
 
-    override fun performWorkAfterSuccessfulCompilation(caches: IncrementalJvmCachesManager) {
+    override fun performWorkAfterSuccessfulCompilation(caches: IncrementalJvmCachesManager, wasIncremental: Boolean) {
         if (classpathChanges is ClasspathChanges.ClasspathSnapshotEnabled) {
             reporter.measure(BuildTime.SHRINK_AND_SAVE_CURRENT_CLASSPATH_SNAPSHOT_AFTER_COMPILATION) {
                 shrinkAndSaveClasspathSnapshot(
-                    classpathChanges, caches.lookupCache, currentClasspathSnapshot, shrunkCurrentClasspathAgainstPreviousLookups,
-                    ClasspathSnapshotBuildReporter(reporter)
+                    wasIncremental, classpathChanges, caches.lookupCache, currentClasspathSnapshot,
```

Comment:
Can you pass `classpathChanges = NotAvailableForNonIncrementalRun`? Shrinker should handle this automatically?

Reply:
Right, as we've discussed offline, that would keep the logic in the shrinker smaller, but keeping the `classpathChanges` variable unchanged and separating it from and compilation incrementality is also nice,  so I'll keep this code then.